src/plugins/navigation_embeddable/public/components/external_link/external_link_component.test.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import userEvent from '@testing-library/user-event';
+import { createEvent, fireEvent, render, screen } from '@testing-library/react';
+import {
+  NavigationEmbeddable,
+  NavigationEmbeddableContext,
+} from '../../embeddable/navigation_embeddable';
+import { mockNavigationEmbeddable } from '../../../common/mocks';
+import { NAV_VERTICAL_LAYOUT } from '../../../common/content_management';
+import { ExternalLinkComponent } from './external_link_component';
+import { coreServices } from '../../services/kibana_services';
+import { DEFAULT_URL_DRILLDOWN_OPTIONS } from '@kbn/ui-actions-enhanced-plugin/public';
+
+describe('external link component', () => {
+  const defaultLinkInfo = {
+    destination: 'https://example.com',
+    order: 1,
+    id: 'foo',
+    type: 'externalLink' as const,
+  };
+
+  let navEmbeddable: NavigationEmbeddable;
+  beforeEach(async () => {
+    window.open = jest.fn();
+    navEmbeddable = await mockNavigationEmbeddable({});
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('by default opens in new tab', async () => {
+    render(
+      <NavigationEmbeddableContext.Provider value={navEmbeddable}>
+        <ExternalLinkComponent link={defaultLinkInfo} layout={NAV_VERTICAL_LAYOUT} />
+      </NavigationEmbeddableContext.Provider>
+    );
+
+    const link = await screen.findByTestId('externalLink--foo');
+    expect(link).toBeInTheDocument();
+    await userEvent.click(link);
+    expect(window.open).toHaveBeenCalledWith('https://example.com', '_blank');
+  });
+
+  test('modified click does not trigger event.preventDefault', async () => {
+    const linkInfo = {
+      ...defaultLinkInfo,
+      options: { ...DEFAULT_URL_DRILLDOWN_OPTIONS, openInNewTab: false },
+    };
+    render(
+      <NavigationEmbeddableContext.Provider value={navEmbeddable}>
+        <ExternalLinkComponent link={linkInfo} layout={NAV_VERTICAL_LAYOUT} />
+      </NavigationEmbeddableContext.Provider>
+    );
+    const link = await screen.findByTestId('externalLink--foo');
+    expect(link).toHaveTextContent('https://example.com');
+    const clickEvent = createEvent.click(link, { ctrlKey: true });
+    const preventDefault = jest.spyOn(clickEvent, 'preventDefault');
+    fireEvent(link, clickEvent);
```

Comment:
This test logs a console error like `Error: Not implemented: navigation (except hash changes)`. This is expected since jsdom [does not handle navigation](https://github.com/jsdom/jsdom/issues/2112). The tests pass, but the error is annoying. Mocking the `window.location` methods did not resolve it. I think the way to fix this is to call `preventDefault`, but we need to check that it isn't called. 😖

Reply:
Makes sense to me! Maybe we should add a comment describing this too? :eyes: